src/test/munit/xa/activemq-xa-other-broker-test-case.xml

```diff
@@ -0,0 +1,225 @@
+                </try>
+                <error-handler>
+                    <on-error-continue/>
+                </error-handler>
+            </try>
+        </munit:execution>
+        <munit:validation>
+            <try>
+                <jms:consume config-ref="JMS_Config_XA_MAX_DOCKER_BROKER" destination="someDest"/>
+                <munit-tools:fail message="Operation should have been rolled back"/>
+                <error-handler>
+                    <on-error-continue type="JMS:TIMEOUT"/>
+                </error-handler>
+            </try>
+            <try>
+                <jms:consume config-ref="JMS_Config_XA" destination="someOtherDest"/>
+                <munit-tools:fail message="Operation should have been rolled back"/>
+                <error-handler>
+                    <on-error-continue type="JMS:TIMEOUT"/>
+                </error-handler>
+            </try>
+        </munit:validation>
+    </munit:test>
+
+    <munit:test name="commit-all">
+        <munit:execution>
+            <try>
+                <try transactionType="XA" transactionalAction="ALWAYS_BEGIN">
+                    <set-payload value="first"/>
+                    <flow-ref name="publish-on-docker-broker"/>
+                    <set-payload value="second"/>
+                    <flow-ref name="publish-on-memory"/>
+                    <raise-error type="APP:SOME"/>
+                    <error-handler>
+                        <on-error-continue/>
+                    </error-handler>
+                </try>
+                <error-handler>
+                    <on-error-continue/>
+                </error-handler>
+            </try>
+        </munit:execution>
+        <munit:validation>
+            <jms:consume config-ref="JMS_Config_XA_MAX_DOCKER_BROKER" destination="someDest"/>
+            <munit-tools:assert-that expression="#[payload]" is="#[MunitTools::equalTo('first')]"/>
+            <jms:consume config-ref="JMS_Config_XA" destination="someOtherDest"/>
+            <munit-tools:assert-that expression="#[payload]" is="#[MunitTools::equalTo('second')]"/>
+        </munit:validation>
+    </munit:test>
+
+    <munit:test name="second-operation-fails">
+        <munit:execution>
+            <try>
+                <try transactionType="XA" transactionalAction="ALWAYS_BEGIN">
+                    <set-payload value="first"/>
+                    <flow-ref name="publish-on-memory"/>
+                    <set-payload value="second"/>
+                    <flow-ref name="publish-on-docker-broker-illegal-body"/>
+                    <error-handler>
+                        <on-error-continue type="JMS:ILLEGAL_BODY"/>
+                    </error-handler>
+                </try>
+                <error-handler>
+                    <on-error-continue/>
+                </error-handler>
+            </try>
+        </munit:execution>
+        <munit:validation>
+            <jms:consume config-ref="JMS_Config_XA" destination="someOtherDest"/>
```

Comment:
shouldn't the publish of this message have been rolled back too?

Reply:
No, since the error was handled by an on-error-continue -> transaction was committed.